### ISSUE TITLE
Execution of embedded linux image failed: Out of Resources - Dell T610

### ISSUE THREAD

**Issue by abclution** [zbm-dev/zfsbootmenu#293]
So trying to get zfsbootmenu working and I'm getting this message. 
Reports for this error (internet wide) are very scarce.

Execution of embedded linux image failed: Out of Resources

![image](https://user-images.githubusercontent.com/10130237/164294341-83918452-975e-4f58-9920-cbdd6ca18aca.png)

Seems that there is something incompatible here. 
I am using the portable method (https://github.com/zbm-dev/zfsbootmenu/wiki/Portable-ZFSBootMenu) to test this out before I set it up permanently and so far no go. 

I have another machine which boots with zbm just fine. Any hints?

**Comment by zdykstra** [comment]
It looks like you're attached to a BMC/iDrac of some kind. What is the underlying physical hardware - CPU, memory, etc ? If you can't directly load the EFI image from your firmware, you can try to use an intermediary like rEFInd to load it for you.

**Comment by abclution** [comment]
Yes, correct iDrac. Shes an older beast but a good one.

Dell T610, whatever motherboard it has 
~128 gigs of ram
Dual Xeon L5640

I will try reFind next was just hoping it was something simple

**Comment by zdykstra** [comment]
I'm able to boot a Dell R710 with rEFInd loading a ZBM image. I haven't tried direct booting on that machine. Dell UEFI from that era is ... not great. The only suggestion I have would be to swap out the UEFI stub loader that's embedded in the EFI file with the newest one from systemd sources. That could potentially help you, but you'd have to build the image / source the stub loader on your own.

**Comment by abclution** [comment]
Alright, I'll give it a poke. Been reading about zfsbootmenu since it was announced 1+ years ago but never had a chance to try it. It really is the "missing piece" for root on zfs. :)

**Comment by abclution** [comment]
Hmm. Same error using ReFind.. 

![image](https://user-images.githubusercontent.com/10130237/164323430-d98d5183-c219-4ae6-b246-f8fa385250e9.png)

**Comment by zdykstra** [comment]
rEFInd will also boot the component builds. They're the exact same parts that make up the EFI release, without the stub loader. You should be able to extract them in the same place as the v1.12.0.EFI and have rEFInd locate them. It might be slightly confused with an EFI file there already, so you might need to remove that first.

**Comment by abclution** [comment]
Sorry you lost me there? Component builds? 
You don't mean the zfsbootmenu-recovery-vmlinuz-x86_64-v1.12.0.EFI recovery image right? 

Also what is the recovery image anyways?

**Comment by zdykstra** [comment]
> Sorry you lost me there? Component builds? You don't mean the zfsbootmenu-recovery-vmlinuz-x86_64-v1.12.0.EFI recovery image right?
> 
> Also what is the recovery image anyways?

The component builds are the `.tar.gz` archives attached to the releases. They're a paired kernel and initramfs that can be booted by generally any initial stage bootloader (syslinux, rEFInd, Grub if you're feeling like you hate life). The latest release is available at https://github.com/zbm-dev/zfsbootmenu/releases/download/v1.12.0/zfsbootmenu-release-x86_64-v1.12.0.tar.gz or at https://get.zfsbootmenu.org/ .

The recovery images contain a few additional tools - namely things related to disk partitioning and EFI variable configuration.

**Comment by abclution** [comment]
Ah ok. Got you.
Yeah I'm gonna try building one like you mentioned before.. and replace the stub.
Its new territory for me so gonna take me a bit to figure out what I need to do.

**Comment by abclution** [comment]
Ok, the component build seems to work... I think.
I lost monitor signal, during I assume the menu option so I didn't see/get a choice of roots to boot from, but it did boot into my default root after. 

This server takes almost 4 minutes to reboot... molasses

**Comment by abclution** [comment]
Yup, it works with the component build, although the monitor does spaz out for quite a bit with out of range error and the zbm menu looks a bit more bare than the screenshots not sure if its related or pertinent, but it seems to work! Now I actually gotta try booting to another root!

Thanks for your assistance @zdykstra 

![image](https://user-images.githubusercontent.com/10130237/164333234-fd6b516f-ebb8-45bb-b96e-672e5b007367.png)

**Comment by zdykstra** [comment]
That screenshot looks accurate. The help text on the bottom is complete. The header lists the boot environment and the kernel in it, and then the arguments it will pass to the kernel. With only one boot environment, there just isn't a lot to show.

**Comment by acidvegas** [comment]
Reporting here that I had the same issue on a poweredge r620. Same errors. Nothing works....would love to see the developers solve this riddle without having to do some weird rEFInd hack. Until then, I cant even use this shit....

**Comment by zdykstra** [comment]
> Reporting here that I had the same issue on a poweredge r620. Same errors. Nothing works....would love to see the developers solve this riddle without having to do some weird rEFInd hack. Until then, I cant even use this shit....

The documented solution is to enable component builds and have `rEFInd` load/boot a bare kernel and initramfs.
